x-pack/filebeat/input/httpinput/input.go

```diff
@@ -0,0 +1,329 @@
+	httpRequest  http.Request            // Current Request
+	httpResponse http.ResponseWriter     // Current ResponseWriter
+	eventObject  *map[string]interface{} // Current event object
+}
+
+// NewInput creates a new httpjson input
+func NewInput(
+	cfg *common.Config,
+	connector channel.Connector,
+	inputContext input.Context,
+) (input.Input, error) {
+	// Extract and validate the input's configuration.
+	conf := defaultConfig()
+	if err := cfg.Unpack(&conf); err != nil {
+		return nil, err
+	}
+	// Build outlet for events.
+	out, err := connector.ConnectWith(cfg, beat.ClientConfig{
+		Processing: beat.ProcessingConfig{
+			DynamicFields: inputContext.DynamicFields,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Wrap input.Context's Done channel with a context.Context. This goroutine
+	// stops with the parent closes the Done channel.
+	inputCtx, cancelInputCtx := context.WithCancel(context.Background())
+	go func() {
+		defer cancelInputCtx()
+		select {
+		case <-inputContext.Done:
+		case <-inputCtx.Done():
+		}
+	}()
+
+	// If the input ever needs to be made restartable, then context would need
+	// to be recreated with each restart.
+	workerCtx, workerCancel := context.WithCancel(inputCtx)
+
+	in := &HttpInput{
+		config:       conf,
+		log:          logp.NewLogger("httpinput"),
+		outlet:       out,
+		inputCtx:     inputCtx,
+		workerCtx:    workerCtx,
+		workerCancel: workerCancel,
+	}
+
+	in.log.Info("Initialized httpinput input.")
+	return in, nil
+}
+
+// Run starts the input worker then returns. Only the first invocation
+// will ever start the worker.
+func (in *HttpInput) Run() {
+	in.workerOnce.Do(func() {
+		in.workerWg.Add(1)
+		go func() {
+			in.log.Info("httpinput worker has started.")
+			defer in.log.Info("httpinput worker has stopped.")
+			defer in.workerWg.Done()
+			defer in.workerCancel()
+			if err := in.run(); err != nil {
+				in.log.Error(err)
+				return
+			}
+		}()
+	})
+}
+
+func (in *HttpInput) run() error {
+	var err error
+	// Create worker context
+	ctx, cancel := context.WithCancel(in.workerCtx)
+	defer cancel()
+
+	// Initialize the HTTP server
+	err = in.createServer()
+
+	if err != nil && err != http.ErrServerClosed {
+		in.log.Fatalf("HTTP Server could not start, error: %v", err)
+	}
+
+	// Infinite Loop waiting for agent to stop
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		}
+	}
+	return err
+}
+
+// Stops HTTP input and waits for it to finish
+func (in *HttpInput) Stop() {
+	in.httpServer.Shutdown(in.workerCtx)
+	in.workerCancel()
+	in.workerWg.Wait()
+}
+
+// Wait is an alias for Stop.
+func (in *HttpInput) Wait() {
+	in.Stop()
+}
+
+func (in *HttpInput) createServer() error {
+	// Merge listening address and port
+	var address strings.Builder
+	address.WriteString(in.config.ListenAddress + ":" + in.config.ListenPort)
+
+	in.httpMux = http.NewServeMux()
+	in.httpMux.HandleFunc(in.config.URL, in.apiResponse)
+
+	if in.config.UseSSL == true {
```

Comment:
Please rewrite the following part as `in.config.UseSSL == true` is redundant.
```golang
	if in.config.UseSSL {
		in.httpServer = &http.Server{Addr: address.String(), Handler: in.httpMux}
		return in.httpServer.ListenAndServeTLS(in.config.SSLCertificate, in.config.SSLKey)
	}
	in.httpServer = &http.Server{Addr: address.String(), Handler: in.httpMux}
	return in.httpServer.ListenAndServe()
```

Also, please remove the following line because it is dead code. `in.config.UseSSL` is either `true` or `false`. There is no third value for `bool`. So this line is never reached.
```golang
	return errors.New("SSL settings missing")
```